### ISSUE TITLE
Switching to the new lightning trainer

### ISSUE THREAD

**Issue by dcfidalgo** [recognai/biome-text#543]
This is an issue to keep track of the process of switching to the new lightning trainer:
 
- [x] Basic training with lightning trainer
- [x] Configure default logging (wandb, tensorboard, csv) 
- [x] Configure default model checkpointing, loading best weights in the end #541 #542
- [x] Save trained pipeline to an output dir after training #542 
- [x] Add default early stopping #544 
- [x] Use GPU automatically if available #545 
- [x] Add default learning rate scheduler #546 
- [x] Adapt `TuneExperiment` to use new trainer #548 
- [x] Add best_metrics #550 
- [x] Adapt documentation/tutorials #551

**Comment by dcfidalgo** [comment]
I'm waiting for the 1.3.0 release of Pytorch Lightning and then i will make a release that supports both trainers (AllenNLP's and PL's). After this release i will remove the AllenNLP Trainer and bumb up the AllenNLP version.
